cirq/value/duration.py

```diff
@@ -13,120 +13,234 @@
+            *,  # Force keyword args.
+            picos: Union[int, float, sympy.Basic] = 0,
+            nanos: Union[int, float, sympy.Basic] = 0,
+            micros: Union[int, float, sympy.Basic] = 0,
+            millis: Union[int, float, sympy.Basic] = 0) -> None:
+        """Initializes a Duration with a time specified in some unit.
 
-        If both picos and nanos are specified, their contributions are added.
+        If multiple arguments are specified, their contributions are added.
 
         Args:
+            value: A value with a pre-specified time unit. Currently only
+                supports 0 and `datetime.timedelta` instances.
             picos: A number of picoseconds to add to the time delta.
             nanos: A number of nanoseconds to add to the time delta.
+            micros: A number of microseconds to add to the time delta.
+            millis: A number of milliseconds to add to the time delta.
+
+        Examples:
+            >>> print(cirq.Duration(nanos=100))
+            100 ns
+            >>> print(cirq.Duration(micros=1.5 * sympy.Symbol('t')))
+            (1500.0*t) ns
         """
-
-        if picos and nanos:
-            self._picos = picos + nanos * 1000
-        else:
-            # Try to preserve type information.
-            self._picos = nanos * 1000 if nanos else picos
+        if value is not None and value != 0:
+            if isinstance(value, datetime.timedelta):
+                # timedelta has microsecond resolution.
+                micros += int(value / datetime.timedelta(microseconds=1))
+            elif isinstance(value, Duration):
+                picos += value._picos
+            else:
+                raise TypeError(f'Not a `cirq.DURATION_LIKE`: {repr(value)}.')
+
+        self._picos: Union[float, int, sympy.Basic] = (picos + nanos * 1000 +
+                                                       micros * 1000_000 +
+                                                       millis * 1000_000_000)
 
     @classmethod
-    def create(cls, duration: Union['Duration', timedelta]) -> 'Duration':
+    @deprecated(deadline='v0.7',
+                fix='Use `cirq.Duration(...)` instead.',
+                func_name='cirq.Duration.create')
+    def create(cls, duration: DURATION_LIKE) -> 'Duration':
         """Creates a Duration from datetime.timedelta if necessary"""
-        if isinstance(duration, cls):
-            return duration
-        if isinstance(duration, timedelta):
-            duration_in_picos = duration.total_seconds() * 10**12
-            return cls(picos=duration_in_picos)
+        return Duration(duration)
 
-        raise TypeError(
-            'Only datetime.timedelta and cirq.Duration are supported.')
+    def _is_parameterized_(self):
+        from cirq import protocols
+        return protocols.is_parameterized(self._picos)
 
-    def total_picos(self) -> float:
+    def _resolve_parameters_(self, param_resolver):
+        from cirq import protocols
```

Comment:
same as above

Reply:
I thought there'd be a circular reference, but upon testing there isn't. Removed.